extensions/google-java/src/main/kotlin/com/connectrpc/extensions/GoogleJavaJSONAdapter.kt

```diff
@@ -16,54 +16,54 @@ package com.connectrpc.extensions
 
 import com.connectrpc.CODEC_NAME_JSON
 import com.connectrpc.Codec
-import com.google.protobuf.GeneratedMessageV3
 import com.google.protobuf.Internal
+import com.google.protobuf.Message
+import com.google.protobuf.TypeRegistry
 import com.google.protobuf.util.JsonFormat
 import okio.Buffer
 import okio.BufferedSource
 import okio.ByteString.Companion.encodeUtf8
 import kotlin.reflect.KClass
+import kotlin.reflect.cast
 
 /**
  * Adapter for Connect to use Google's protobuf-java runtime for
  * deserializing and serializing data types.
  */
-internal class GoogleJavaJSONAdapter<E : GeneratedMessageV3>(
-    clazz: KClass<E>,
+internal class GoogleJavaJSONAdapter<E : Message>(
+    private val clazz: KClass<E>,
+    private val registry: TypeRegistry,
 ) : Codec<E> {
-    /**
-     * Casting assumes the user is using Google's GeneratedMessageV3 type.
-     */
-    @Suppress("UNCHECKED_CAST")
```

Comment:
I got rid of unnecessary suppressions. In some cases (like this one), no unchecked cast was even needed/useful. In other places, we can use `clazz.cast` to perform an actual checked cast.